walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,120 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ListIterator;
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> potentialTimes = new ArrayList<>();
+    potentialTimes.add(TimeRange.WHOLE_DAY);
+    Collection<String> meetingGuests = request.getAttendees();
+    Collection<String> optionalGuests = request.getOptionalAttendees();
+    long meetingDuration = request.getDuration();
+
+    Collection<TimeRange> times = new ArrayList<>();
+
+    ArrayList<TimeRange> badTimes = new ArrayList<>();
+
+    for (Event event: events) {
+      TimeRange eventTime = event.getWhen();
+      Set<String> guests = event.getAttendees();
+      for (String optional: optionalGuests) {
+        if (guests.contains(optional)) {
+          badTimes.add(eventTime);
```

Comment:
Is this saying "for each of the optional attendees, if they're attending current event, then add the current event to the list of bad times"? I'm having trouble tracking the different guests. If so, is `badTimes` just the times that optional attendees can't attend?